### PR TITLE
📝 Clarify SEO visibility setting

### LINKED ISSUE #2488: 📝 Improve search engine visibility page
### Where on docs.platform.sh should be changed?

https://docs.platform.sh/environments/search-engine-visibility.html#how-itrsquos-done

### What exactly should be updated?

Make it clearer that we can't actually hide sites, we just ask engines nicely to ignore them.

Also, the interaction between various settings/configurations could be clearer. For example, if you serve a `robots.txt` file with `disallow *`, it doesn't matter whether the setting to hide is on or off.

`robots.txt` files can be present in the root or with a passthru that serves it.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CollierCZ**
## Why

Closes #2488

## What's changed

Clarified what is set and how it can be overridden

### CODE REVIEW

File: docs/src/environments/search-engine-visibility.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ toc: false
 When you have development environments,
 you don't want search engines indexing them and diluting the SEO of your production site.
 
-All development environments are hidden.
+Search engine indexers are told to ignore all development environments.
```
Comment (trolologuy) [inline docs/src/environments/search-engine-visibility.md:11]: :bulb:
Would it make sense to add a note that not all bots comply to that rule and if privacy is needed it's best to use http auth?

File: docs/src/environments/search-engine-visibility.md [hunk 2]
```diff
@@ -47,14 +49,20 @@ platform environment:info -e <ENVIRONMENT_NAME> restrict_robots true
...
+Search engines are turned away from environments by including a `X-Robots-Tag` header:
 
 ```txt
 X-Robots-Tag: noindex, nofollow
 ```
 
-That tells search engines to not index these sites and not traverse links from these sites.
-This keeps non-production sites out of search engine indexes.
+That tells search engine indexers to not index these sites and not traverse links from these sites.
+This helps keeps non-Production sites out of search engine indexes.
```
Comment (trolologuy) [inline docs/src/environments/search-engine-visibility.md:59]: :lady_beetle:
Typo?
```suggestion
This helps to keep non-Production sites out of search engine indexes.
```

**Review by trolologuy — CHANGES_REQUESTED** [review]

File: docs/src/environments/search-engine-visibility.md (hunk 2 above)
Comment (CollierCZ) [reply docs/src/environments/search-engine-visibility.md:59]: Yes, but `to keep` there is British English, so I'll go with just `keep`.

Comment (CollierCZ) [inline docs/src/environments/search-engine-visibility.md:59]:
```suggestion
This helps keep non-Production sites out of search engine indexes.
```

**Review by trolologuy — APPROVED** [review]
✅

**Pull request merged**